packages/monitor-v2/src/monitor-polymarket/common.ts

```diff
@@ -117,135 +66,120 @@ interface PolymarketOrderBook {
 }
 
 export type Order = { price: number; size: number }[];
-export interface MarketOrderbooks {
-  orderBooks: { bids: Order; asks: Order }[];
-}
 
-export interface StoredNotifiedProposal {
-  txHash: string;
-  question: string;
-  proposedPrice: string;
-  notificationTimestamp: number;
-  requestTimestamp?: string;
-  notified?: boolean;
+export interface MarketOrderbook {
+  bids: Order;
+  asks: Order;
 }
 
-export interface FormattedProposePriceEvent {
-  txHash: string;
+export interface OptimisticPriceRequest {
+  requestHash: string;
+  requestTimestamp: BigNumber;
+  requestLogIndex: number;
   requester: string;
-  proposer: string;
-  timestamp: string;
-  eventTimestamp: number;
-  eventBlockNumber: number;
-  expirationTimestamp: number;
-  proposalTimestamp: number;
-  identifier: string;
   ancillaryData: string;
-  proposedPrice: string;
-  eventIndex: number;
-  oracleAddress: string;
+  requestBlockNumber: number;
+  proposedPrice: BigNumber;
+  proposalTimestamp: BigNumber;
+  proposalHash: string;
+  proposalExpirationTimestamp: BigNumber;
+  proposalLogIndex: number;
 }
 
-export const formatPriceEvents = async (events: ProposePriceEvent[]): Promise<FormattedProposePriceEvent[]> => {
-  const ooDefaultLiveness = 7200;
-  return Promise.all(
-    events.map(async (event: ProposePriceEvent) => {
-      const block = await event.getBlock();
+interface StoredNotifiedProposal {
+  proposalHash: string;
+}
+
+export const getPolymarketProposedPriceRequestsOO = async (
+  params: MonitoringParams,
+  version: "v1" | "v2",
+  requesterAddresses: string[]
+): Promise<OptimisticPriceRequest[]> => {
+  const currentBlockNumber = await params.provider.getBlockNumber();
+  const oneDayInBlocks = POLYGON_BLOCKS_PER_HOUR * 24;
+  const startBlockNumber = currentBlockNumber - oneDayInBlocks;
+  const maxBlockLookBack = params.maxBlockLookBack;
+
+  const searchConfig = {
+    fromBlock: startBlockNumber,
+    toBlock: currentBlockNumber,
+    maxBlockLookBack,
+  };
+
+  const oo = await getContractInstanceWithProvider<OptimisticOracleEthers | OptimisticOracleV2Ethers>(
+    version == "v1" ? "OptimisticOracle" : "OptimisticOracleV2",
+    params.provider
+  );
+
+  const events = await paginatedEventQuery<ProposePriceEvent>(
+    oo,
+    oo.filters.ProposePrice(null, null, null, null, null, null, null, null),
+    searchConfig
+  );
+
+  return events
+    .filter((event) => requesterAddresses.map((r) => r.toLowerCase()).includes(event.args.requester.toLowerCase()))
+    .filter((event) => event.args.expirationTimestamp.gt(BigNumber.from(Math.floor(Date.now() / 1000))))
```

Comment:
is this the correct time to be filtering?

Reply:
Yes, this retains only the proposals that have an expiration timestamp in the future, meaning they can be disputed and are in their liveness period. The timestamps in the contracts are in seconds; that's why we need to do Date.now()/1000.